public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -296,7 +296,7 @@ function getDataLinks(options: ExemplarTraceIdDestination): DataLink[] {
 
   if (options.url) {
     dataLinks.push({
-      title: `Go to ${options.url}`,
+      title: options.urlDisplayLabel || `Go to ${options.url}`,
```

Comment:
You'll also need to add `options.urlDisplayLabel || ` to L287 so that internal links use the custom label (if there is one).